File: Aula06/Exercícios/Mario Desafio/mario.py
from cs50 import get_int


def main():
    a = get_altura()
    for i in range(a):
        # itera os espaços pra inverter a escada
        for j in range(a - 1, i, -1):
            print(" ", end="")
        # itera os blocos na escada
        for l in range(0, i + 1, 1):
            print("#", end="")
        print("  ", end="")
        # itera a segunda escada
        for k in range(0, i + 1, 1):
            print("#", end="")
        print()


# Garante a altura da escada entre 1 e 8 blocos
def get_altura():
    while True:
        n = get_int("Altura da escada: ")
        if n >= 1 and n <= 8:
            break
    return n


main()

File: Week01/Exercises/Mario More/mario.c
//Criar um programa que peça um número ao usuário
//e então criar escadas espelhadas com este número
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    //n = altura desejada, a = degrau, l = largura, s = espaços em branco
    int n = 0, a, l, s;
    do
    {
        n = get_int("Determine a altura da escada: ");
    }
    while (n < 1 || n > 8);

    //criar tijolos
    for (a = 0; a < n; a++)
    {
        //espelhando a primeira escada
        for (s = n - 1; s > a; s--)
        {
            printf(" ");
        }
        for (l = 0; l <= a; l++)
        {
            printf("#");
        }

        //criando a segunda escada
        printf("  ");

        for (l = 0; l <= a; l++)
        {
            printf("#");
        }
        printf("\n");
    }
}